src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -0,0 +1,235 @@
+{
+    public override string M() => null;
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,28): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override string M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(8, 28)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_02()
+        {
+            var source = @"
+class Base
+{
+    public virtual T M<T, U>() where T : class where U : class, T => null;
+}
+class Derived : Base
+{
+    public override U M<T, U>() => null;
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override U M<T, U>() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(8, 23)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_03()
+        {
+            var source = @"
+class Base<T> where T : class
+{
+    public virtual T M() => null;
+}
+class Derived<T, U> : Base<T> where T : class where U : class, T
+{
+    public override U M() => null;
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override U M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(8, 23)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_04()
+        {
+            var source = @"
+class N { }
+class Base
+{
+    public virtual N M() => null;
+}
+class Derived<T> : Base where T : N
+{
+    public override T M() => null;
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (9,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override T M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(9, 23)
```

Comment:
>ERR_FeatureInPreview [](start = 37, length = 20)

Please add LangVer tests for other overrides scenarios (not just methods).

Reply:
This already tests methods, properties and indexers.  Will add events (for which no change of permitted behavior is planned).

---
In reply to: [413972645](https://github.com/dotnet/roslyn/pull/43576#discussion_r413972645) [](ancestors = 413972645)